webapp/platform/client/src/client4.ts

```diff
@@ -3895,6 +3896,14 @@ export default class Client4 {
         );
     };
 
+
+    confirmSelfHostedExpansion = (setupIntentId: string, expandRequest: SelfHostedExpansionRequest) => {
+        return this.doFetch<SelfHostedSignupSuccessResponse>(
+            `${this.getHostedCustomerRoute()}/confirm-expand`,
```

Comment:
Looks like we're missing adding this route. I get this when trying to expand locally:
```
λ~/w/mattermost-server/webapp/channelsλ {"timestamp":"2023-04-18 14:43:06.620 -05:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:179","method":"POST","url":"/api/v4/hosted_customer/customer","request_id":"i97dtf19opdhi8rz7zthzjoner","status_code":"200"}
{"timestamp":"2023-04-18 14:43:08.145 -05:00","level":"debug","msg":"not found handler triggered","caller":"web/web.go:63","path":"/api/v4/hosted_customer/confirm-expand","code":404,"ip":"127.0.0.1"}
```

Reply:
The route is in a separate pr: https://github.com/mattermost/mattermost-server/pull/22657/files#diff-d87ac22b463e8d5b6a9da02c2241564d3c579b4bb9355092d1581e2ff54e4681R35

Reply:
Here's a PR for testing purposes with the front-end and backend merged together: https://github.com/mattermost/mattermost-server/pull/23017

I just separated them because I expected more changes in the back-end initially.

Reply:
Awesome, thanks! Tried them together and seems like it works. Was able to expand.